tooling/reproducible/linux_repro_build_compare.sh

```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```

Comment:
Do we have to move to bash?

Reply:
Using /bin/sh linter isn't happy and I'm not sure why /bin/sh was initially used. I've searched all shell scripts in all adoptium repos and found all shell scripts are using /bin/bash. So I did the change.